rotkehlchen/exchanges/kucoin.py

```diff
@@ -523,47 +533,49 @@ def _deserialize_accounts_balances(
     def _deserialize_asset_movement(
             raw_result: dict[str, Any],
             case: Literal[KucoinCase.DEPOSITS, KucoinCase.WITHDRAWALS],
-    ) -> AssetMovement:
+    ) -> list[AssetMovement]:
         """Process an asset movement result and deserialize it
 
         May raise:
         - DeserializationError
         - UnknownAsset
         - UnsupportedAsset
         """
+        event_type: Literal[HistoryEventType.DEPOSIT, HistoryEventType.WITHDRAWAL]
         if case == KucoinCase.DEPOSITS:
-            category = AssetMovementCategory.DEPOSIT
+            event_type = HistoryEventType.DEPOSIT
         elif case == KucoinCase.WITHDRAWALS:
-            category = AssetMovementCategory.WITHDRAWAL
+            event_type = HistoryEventType.WITHDRAWAL
         else:
             raise AssertionError(f'Unexpected case: {case}')
 
         try:
-            timestamp_ms = deserialize_timestamp(raw_result['createdAt'])
-            timestamp = Timestamp(int(timestamp_ms / 1000))
+            timestamp = TimestampMS(deserialize_timestamp(raw_result['createdAt']))
             address = raw_result['address']
             # The transaction id can have an @ which we should just get rid of
             transaction_id = raw_result['walletTxId'].split('@')[0]
             amount = deserialize_asset_amount(raw_result['amount'])
             fee = deserialize_fee(raw_result['fee'])
             fee_currency_symbol = raw_result['currency']
-            link_id = raw_result.get('id', '')  # NB: id only exists for withdrawals
+            unique_id = raw_result.get('id')  # NB: id only exists for withdrawals
         except KeyError as e:
             raise DeserializationError(f'Missing key: {e!s}.') from e
 
         fee_asset = asset_from_kucoin(fee_currency_symbol)
 
-        return AssetMovement(
+        return create_asset_movement_with_fee(
             timestamp=timestamp,
             location=Location.KUCOIN,
-            category=category,
-            address=address,
-            transaction_id=transaction_id,
+            event_type=event_type,
             asset=fee_asset,
             amount=amount,
             fee_asset=fee_asset,
             fee=fee,
-            link=link_id,
+            unique_id=unique_id or transaction_id,
```

Comment:
unique_id as per its name should have the  guarantee of uniqueness for the given exchange. Transaction id should probably also have the same uniqueness guarantee but are you sure?

Reply:
As far as I can tell from the docs (https://www.kucoin.com/docs/rest/funding/deposit/get-deposit-list), the value in `transaction_id` (from `walletTxId`) will be an onchain tx hash, which I expect should always be unique.